src/overlays/actors/ovl_En_Zf/z_en_zf.c

```diff
@@ -82,115 +151,2291 @@ static ColliderQuadInit D_80B4A204 = {
+void EnZf_SetupSlash(EnZf* this) {
+    Animation_Change(&this->skelAnime, &gZfSlashAnim, 1.25f, 0.0f, Animation_GetLastFrame(&gZfSlashAnim), 2, -4.0f);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zf/func_80B44DC4.s")
+    if (this->actor.params == ENZF_TYPE_DINOLFOS) {
+        this->skelAnime.playSpeed = 1.75f;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zf/func_80B44E8C.s")
+    this->swordCollider.base.atFlags &= ~AT_BOUNCED;
+    this->action = ENZF_ACTION_SLASH;
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_RIZA_CRY);
+    this->actor.speedXZ = 0.0f;
+    EnZf_SetupAction(this, EnZf_Slash);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zf/func_80B450AC.s")
+void EnZf_Slash(EnZf* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    s16 rotDiff;
+    s16 yawDiff;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zf/func_80B45174.s")
+    this->actor.speedXZ = 0.0f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zf/func_80B45384.s")
+    if ((s32)this->skelAnime.curFrame == 10) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_RIZA_ATTACK);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zf/func_80B4543C.s")
+    if (SkelAnime_Update(&this->skelAnime)) {
+        EffectBlure_AddSpace(Effect_GetByIndex(this->blureIndex));
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zf/func_80B456B4.s")
+        if ((this->actor.params == ENZF_TYPE_DINOLFOS) && !Actor_IsFacingPlayer(&this->actor, 5460)) {
+            func_80B45384(this);
+            this->unk_3F0 = Rand_ZeroOne() * 5.0f + 5.0f;
+            this->unk_3F4 = Rand_ZeroOne() * 20.0f + 100.0f;
+        } else if ((Rand_ZeroOne() > 0.7f) || (this->actor.xzDistToPlayer >= 120.0f)) {
+            func_80B45384(this);
+            this->unk_3F0 = Rand_ZeroOne() * 5.0f + 5.0f;
+        } else {
+            this->actor.world.rot.y = this->actor.yawTowardsPlayer;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zf/func_80B45748.s")
+            if (Rand_ZeroOne() > 0.7f) {
+                func_80B483E4(this, globalCtx);
+            } else {
+                rotDiff = player->actor.shape.rot.y - this->actor.shape.rot.y;
+                rotDiff = ABS(rotDiff);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zf/func_80B45E30.s")
+                if (rotDiff <= 10000) {
+                    yawDiff = this->actor.yawTowardsPlayer - this->actor.shape.rot.y;
+                    yawDiff = ABS(yawDiff);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zf/func_80B45EF0.s")
+                    if (yawDiff > 16000) {
+                        this->actor.world.rot.y = this->actor.yawTowardsPlayer;
+                        func_80B483E4(this, globalCtx);
+                    } else if ((player->stateFlags1 & 0x6010)) {
```

Comment:
```suggestion
                    } else if ((player->stateFlags1 & 0x6010) != 0) {
```
I like != 0 when several flags, if not at least remove parens